scripts/ci/build.sh

```diff
@@ -4,6 +4,10 @@ set -ex
 ROOTDIR="$(dirname "$0")/../.."
 cd "${ROOTDIR}"
 
+ABSOLUTE_PATH=$(pwd)
+
+git config --global --add safe.directory "${ABSOLUTE_PATH}"
```

Comment:
Interesting. Looking at the documentation: https://git-scm.com/docs/git-config#Documentation/git-config.txt-safedirectory

This feels a bit scary. This is where the release binaries are built right? I wonder if a git hook can influence the binary somehow?

Reply:
Also, the `--global` flag modifies user's `~/.gitconfig`. I think it would be better to modify config only for the current repo in case someone has to run this script on his machine.

Reply:
> This feels a bit scary. This is where the release binaries are built right? I wonder if a git hook can influence the binary somehow?

After taking a closer look at it, the root of the issue is that we're giving git inside the container a repo dir owned by the user from outside of the container. git notices different users and bails. I don't think there's any inherent risk in that. At least it's not that different from the situation where you just clone the repo and run git on it. It's really the same user doing things.

The thing that's kinda bad is that we're running as root inside the container and it's not a good practice but this is independent of this PR. We already have this problem.

Overall we could work around the issue by chowning the files or by ensuring that we use the same user on both sides. But I think that `safe.directory` fix is good enough. Just IMO the `build.sh` script is not the best place to run it since it's a global change. I think it belongs either in the dockerfile or in `docker_upgrade.sh`. Probably the latter since the repo dir is not hard-coded in the image and it's the script that attaches the volume that determines that location.